service/api/service.go

```diff
@@ -432,56 +433,84 @@ func (s *serviceImpl) ApiV1WorkflowRpcPost(ctx context.Context, req iwfidl.Workf
 
 	var queryResp service.PrepareRpcQueryResponse
 	queryToPrepare := true
+
 	if req.GetUseMemoForDataAttributes() {
 		var searchAttributes []iwfidl.SearchAttribute
 		var dataAttributes []iwfidl.KeyValue
 
-		requestedSAs := req.SearchAttributes
+		var requestedSAs []iwfidl.SearchAttributeKeyAndType
+
 		saPolicy := req.GetSearchAttributesLoadingPolicy()
-		if saPolicy.GetPersistenceLoadingType() != iwfidl.ALL_WITHOUT_LOCKING {
+		switch saPolicy.GetPersistenceLoadingType() {
+		case iwfidl.PARTIAL_WITH_EXCLUSIVE_LOCK:
```

Comment:
we need to return error for this case. this will be supported in https://github.com/indeedeng/iwf/issues/305